prototype_source/backend_config_tutorial.rst

```diff
@@ -0,0 +1,410 @@
+``relu``, and ``linear``. See `this
+example <https://gist.github.com/jerryzh168/8bea7180a8ba3c279f2c9b050f2a69a6>`__
+for a slightly more complicated usage.
+
+set_fused_module
+^^^^^^^^^^^^^^^^
+
+This is used to identify fused weighted modules (e.g.
+``torch.ao.nn.intrinsic.LinearReLU``) that need to be converted to
+reference quantized modules.
+
+Data Type Restrictions
+~~~~~~~~~~~~~~~~~~~~~~
+
+Each DTypeConfig attached to a BackendPatternConfig represents a set of
+supported data types for the input/output activations, weights, and
+biases. These data types are matched against the ones specified in the
+user’s QConfig. If there is a match, and the QConfig satisfies the
+constraints specified in the DTypeConfig (if any), then we will quantize
+the given pattern using this DTypeConfig. Otherwise, the QConfig is
+ignored and the pattern will not be quantized.
+
+There are two ways of specifying ``input_dtype``, ``output_dtype``, and
+``weight_dtype``, as simple ``torch.dtype`` or as
+``DTypeWithConstraints``. The constraints currently supported are:
+
+- **quant_min_lower_bound** and **quant_max_upper_bound**: Lower and upper
+  bounds for the minimum and maximum quantized values respectively. If the
+  QConfig’s ``quant_min`` and ``quant_max`` fall outside this range, then
+  the QConfig will be ignored.
+- **scale_min_lower_bound** and **scale_max_upper_bound**: Lower and
+  upper bounds for the minimum and  aximum scale values respectively. If
+  the QConfig’s minimum scale value (currently exposed as ``eps``) falls
+  below the lower bound, then the QConfig will be ignored. Note that the
+  upper bound is currently not enforced.
+- **scale_exact_match** and **zero_point_exact_match**: Exact match
+  requirements for scale and zero point, to be used for operators with
+  fixed quantization parameters such as sigmoid and tanh. If the observer
+  specified in the QConfig is neither ``FixedQParamsObserver`` nor
+  ``FixedQParamsFakeQuantize``, or if the quantization parameters don't
+  match, then the QConfig will be ignored.
+
+End-to-End Example
+------------------
+
+Here we define a simple example BackendConfig that supports 3 things:
+
+- Quantizing linear to int8 (unsigned)
+- Fusing conv2d followed by relu
+- Quantizing the fused conv2d-relu to int8 (unsigned)
+
+In this section, we will run through an example model with these
+patterns and show how to use this BackendConfig in ``prepare_fx`` and
+``convert_fx``.
+
+.. code:: ipython3
+
+    import torch
+    from torch.ao.quantization import (
+        default_weight_observer,
+        get_default_qconfig_mapping,
+        MinMaxObserver,
+        QConfig,
+        QConfigMapping,
+    )
+    from torch.ao.quantization.backend_config import (
+        BackendConfig,
+        BackendPatternConfig,
+        DTypeConfig,
+        DTypeWithConstraints,
+        ObservationType,
+    )
+    from torch.ao.quantization.fuser_method_mappings import _reverse_sequential_wrapper2
```

Comment:
I remember we talked about not exposing things like `_reverse_sequential_wrapper2` to the user, can we remove this before adding tutorials?

Reply:
Ok, I can inline this

Reply:
> Ok, I can inline this

I would imagine that we actualy remove this from the codebase and expose a more user friendly API, not just in this tutorial